src/init.cpp

```diff
@@ -1476,6 +1477,7 @@ bool AppInitMain(NodeContext& node, interfaces::BlockAndHeaderTipInfo* tip_info)
 
             try {
                 LOCK(cs_main);
+                LOCK(cs_HavePruned);
```

Comment:
Unsure wheter to lock the whole block or line 1485 specifically
My reasoning here was that verifying blocks ought to have higher priority than answering RPC requests